client/src/hooks/useGetData.ts

```diff
@@ -1,39 +1,39 @@
 import { createToken } from 'API/get';
-import { useState, useEffect, useCallback } from 'react';
-import { DailyWeather, HourlyWeather, LocationData, WeatherData } from 'types';
+import LocationContext from 'contexts/LocationContext';
+import { useState, useEffect, useCallback, useContext } from 'react';
+import { DailyWeather, HourlyWeather, LocationData, CurrentWeatherData } from 'types';
 
-const useGetData = (
-  getCurrentWeather,
-  getDailyWeather,
-  getHourlyWeather,
-  getLocation,
-  { coordinates, location }
-) => {
-  const [currentWeather, setCurrentWeather] = useState<WeatherData | null>(null);
+const useGetData = ({ getCurrentWeather, getDailyWeather, getHourlyWeather, getLocation }) => {
+  const { coordinates } = useContext(LocationContext);
+  const [currentWeather, setCurrentWeather] = useState<CurrentWeatherData | null>(null);
   const [locationData, setLocationData] = useState<LocationData | undefined>(undefined);
   const [forecast, setForecast] = useState<DailyWeather[]>([]);
   const [hourlyForecast, setHourlyForecast] = useState<HourlyWeather[]>([]);
   const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [errorMsg, setErrorMsg] = useState<string>('');
 
   const getData = useCallback(async () => {
-    await createToken();
-    const data = await getCurrentWeather(coordinates);
-    const locationResult = await getLocation(location || coordinates);
-    const dailyWeather = await getDailyWeather(coordinates);
-    const hourlyWeather = await getHourlyWeather(coordinates);
-    setCurrentWeather(data);
-    setLocationData(locationResult.locations ? locationResult.locations[0] : locationResult);
-    setForecast(dailyWeather);
-    const first24Hours = hourlyWeather?.slice(0, 24);
-    setHourlyForecast(first24Hours);
-    setIsLoading(false);
+    try {
+      await createToken();
+      const data = await getCurrentWeather(coordinates);
+      const locationResult = await getLocation(coordinates);
+      const dailyWeather = await getDailyWeather(coordinates);
+      const hourlyWeather = await getHourlyWeather(coordinates);
+      setCurrentWeather(data);
+      setLocationData(locationResult.locations ? locationResult.locations[0] : locationResult);
```

Comment:
please be notified that in case `locationResult.locations = []`, you will set `undefined` in state, not `locationResult` value. 
in case this is not an expected behavior, you can modify condition and check length of the array, not just existence of it